packages/template-block/src/components/TemplatePreview.tsx

```diff
@@ -0,0 +1,205 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import {
+    AppBridgeBlock,
+    Template,
+    TemplateLegacy,
+    useBlockAssets,
+    useBlockSettings,
+    useEditorState,
+} from '@frontify/app-bridge';
+import {
+    ActionMenu,
+    Button,
+    ButtonRounding,
+    ButtonSize,
+    ButtonStyle,
+    ButtonType,
+    Color,
+    Flyout,
+    IconDotsVertical,
+    IconPlus20,
+    merge,
+} from '@frontify/fondue';
+import { getRgbaString } from '../utils';
+import { useCallback, useState } from 'react';
+import {
+    Settings,
+    cornerRadiusValues,
+    previewDisplayValues,
+    previewHeightValues,
+    previewImageAnchoringValues,
+    textPositioningToFlexDirection,
+} from '../types';
+import { SETTING_ID } from '../constants';
+
+export type TemplatePreviewProps = {
+    appBridge: AppBridgeBlock;
+    template: Template | null;
+    onUpdateTemplate: (key: string, newTemplateIds: number[]) => Promise<void>;
+    onUpdateTemplateTitle: (value: string) => void;
+    onUpdateTemplateDescription: (value: string) => void;
+    onError: (message: string) => void;
+};
+
+export const TemplatePreview = ({
+    appBridge,
+    template,
+    onUpdateTemplate,
+    onUpdateTemplateTitle,
+    onUpdateTemplateDescription,
+    onError,
+}: TemplatePreviewProps) => {
+    const [blockSettings, updateBlockSettings] = useBlockSettings<Settings>(appBridge);
+    const { blockAssets } = useBlockAssets(appBridge);
+    const isEditing = useEditorState(appBridge);
+
+    const {
+        hasPreviewBackgroundColor,
+        previewBackgroundColor,
+        hasPreviewBorder,
+        previewBorderColor,
+        previewBorderStyle,
+        previewBorderWidth,
+        isPreviewCorderRadiusCustom,
+        previewCornerRadiusSimple,
+        previewCornerRadiusCustom,
+        textRatio,
+        isPreviewHeightCustom,
+        previewHeightSimple,
+        previewHeightCustom,
+        previewDisplay,
+        previewImageAnchoring,
+        textPositioning,
+    } = blockSettings;
+
+    const { previewCustom } = blockAssets;
+
+    const flexDirection = textPositioningToFlexDirection[textPositioning];
+    const isRows = useCallback(() => flexDirection === 'row' || flexDirection === 'row-reverse', [flexDirection]);
+
+    const [isActionFlyoutOpen, setIsActionFlyoutOpen] = useState(false);
+
+    const updateTemplateTitleAndDescription = (template: TemplateLegacy) => {
+        const { title, description } = blockSettings;
+        const newTitleValue = title ? title : template.title;
+        const newDescriptionValue = description ? description : template.description;
+
+        onUpdateTemplateTitle(newTitleValue);
+        onUpdateTemplateDescription(newDescriptionValue);
+    };
+
+    const onTemplateSelected = useCallback(async (result: TemplateLegacy) => {
```

Comment:
Does this need to be a useCallback? Seems like data could get stale after a while especially if the dependency array is empty.

Reply:
You're right. Seems like `useCallback` is useless here 👍